doc/contributing/maintaining-shared-library-support.md

```diff
@@ -0,0 +1,117 @@
+# Maintaining shared library support
+
+Node.js unofficially supports a build option where Node.js is built as
+a shared library (node.dll/libnode.so). The shared library provides
+an easy way to way to embed Node.js into other aplications and to
+have multiple applications use a single copy of Node.js instead
+of having to bundle in the full Node.js footprint into each application.
+
```

Comment:
may be worth mentioning at this point that a great share of node's memory footprint goes into the shared library, and for workloads that require such multiple node instances can greatly benefit from this approach?

Reply:
@gireeshpunathil thanks, added a sentence to that effect.